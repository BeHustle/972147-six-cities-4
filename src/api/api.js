import axios from 'axios';

const URL = `https://4.react.pages.academy/six-cities`;
const TIMEOUT = 5000;

export const createAPI = (dispatch = null) => {
  const api = axios.create({
    baseURL: URL,
    timeout: TIMEOUT,
    withCredentials: true
  });

  const onSuccess = (response) => response;
  const onFail = ({response}) => {
    dispatch();
    throw response;
  };

  api.interceptors.response.use(onSuccess, onFail);
  return api;
};
