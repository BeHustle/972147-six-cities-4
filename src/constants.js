const CARD_TYPES = [
  `Apartment`,
  `Private room`
];

const Cities = {
  AMSTERDAM: [52.38333, 4.9]
};

const MAP_ZOOM = 12;

const ICON_SIZE = [30, 30];

const ICON_PATH = `img/pin.svg`;

export {CARD_TYPES, Cities, MAP_ZOOM, ICON_SIZE, ICON_PATH};
