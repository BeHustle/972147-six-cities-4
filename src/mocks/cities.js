export const cities = [
  {
    id: 1,
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 2,
    name: `Paris`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 3,
    name: `Cologne`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 4,
    name: `Brussels`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 5,
    name: `Hamburg`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 6,
    name: `Dusseldorf`,
    coordinates: [52.38333, 4.9]
  },
];
