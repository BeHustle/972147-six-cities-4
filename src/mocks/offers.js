export const offers = [
  {
    id: 1,
    name: `Beautiful & luxurious apartment at great location`,
    price: 132,
    image: `img/apartment-01.jpg`,
    type: `Private room`,
    isPremium: false,
    inBookmarks: true,
    rating: 4
  },
  {
    id: 2,
    name: `Wood and stone place`,
    price: 254,
    image: `img/apartment-02.jpg`,
    type: `Apartment`,
    isPremium: false,
    inBookmarks: false,
    rating: 5
  },
  {
    id: 3,
    name: `Canal View Prinsengracht`,
    price: 178,
    image: `img/apartment-03.jpg`,
    type: `Private room`,
    isPremium: true,
    inBookmarks: true,
    rating: 3
  },
  {
    id: 4,
    name: `Nice, cozy, warm big bed apartment`,
    price: 995,
    image: `img/room.jpg`,
    type: `Apartment`,
    isPremium: true,
    inBookmarks: false,
    rating: 1
  },
  {
    id: 5,
    name: `Beautiful & luxurious apartment at great location`,
    price: 80,
    image: `img/apartment-01.jpg`,
    type: `Private room`,
    isPremium: false,
    inBookmarks: true,
    rating: 2
  },
];
