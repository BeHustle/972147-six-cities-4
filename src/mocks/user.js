const email = `Oliver.conner@gmail.com`;

export {email};
