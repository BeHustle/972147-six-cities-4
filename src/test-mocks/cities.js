export const cities = [
  {
    id: 1,
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9],
    zoom: 10
  },
  {
    id: 2,
    name: `Paris`,
    coordinates: [48.85875194, 2.33926391],
    zoom: 10
  },
  {
    id: 3,
    name: `Cologne`,
    coordinates: [50.938361, 6.959974],
    zoom: 10
  },
  {
    id: 4,
    name: `Brussels`,
    coordinates: [50.8465573, 4.351697],
    zoom: 10
  },
  {
    id: 5,
    name: `Hamburg`,
    coordinates: [53.550341, 10.000654],
    zoom: 10
  },
  {
    id: 6,
    name: `Dusseldorf`,
    coordinates: [51.2254018, 6.7763137],
    zoom: 10
  },
];
