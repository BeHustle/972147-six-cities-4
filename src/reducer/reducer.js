import {combineReducers} from 'redux';
import {Namespace} from './namespace';
import {reducer as app} from './app/app.reducer';
import {reducer as data} from './data/data.reducer';
import {reducer as user} from './user/user.reducer';

export default combineReducers({
  [Namespace.APP]: app,
  [Namespace.DATA]: data,
  [Namespace.USER]: user
});
