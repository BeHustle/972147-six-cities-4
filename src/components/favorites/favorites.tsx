import * as React from 'react';
import {Link} from 'react-router-dom';
import {getFavoriteCities, getFavoriteOffers} from '../../reducer/data/data.selectors';
import AppRoute from '../../routes';
import EmptyFavoritesScreen from '../empty-favorites-screen/empty-favorites-screen';
import FavoritesList from '../favorites-list/favorites-list';
import Header from '../header/header';
import {connect} from 'react-redux';
import {Operation as DataOperation} from '../../reducer/data/data.reducer';
import {CityInterface, OfferInterface} from "../../types";

interface Props {
  offers: Array<OfferInterface>;
  cities: Array<CityInterface>;
  onFavoritesDidMount: () => void;
  onFavoriteClick: () => void;
}

class Favorites extends React.PureComponent<Props, {}> {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.onFavoritesDidMount();
  }

  render() {
    const {offers, cities, onFavoriteClick} = this.props;
    return <div className={`page ${offers.length || `page--favorites-empty`}`}>

      <Header />

      {offers.length
        ? <main className="page__main page__main--favorites">
          <div className="page__favorites-container container">
            <section className="favorites">
              <h1 className="favorites__title">Saved listing</h1>
              <FavoritesList
                onFavoriteClick={onFavoriteClick}
                offers={offers}
                cities={cities}
              />
            </section>
          </div>
        </main>
        : <EmptyFavoritesScreen />}

      <footer className="footer container">
        <Link to={AppRoute.MAIN} className="footer__logo-link">
          <img className="footer__logo" src="./img/logo.svg" alt="6 cities logo" width="64" height="33" />
        </Link>
      </footer>
    </div>;
  }
}

const mapStateToProps = (state) => ({
  offers: getFavoriteOffers(state),
  cities: getFavoriteCities(state)
});

const mapDispatchToProps = (dispatch) => ({
  onFavoritesDidMount() {
    dispatch(DataOperation.loadFavoriteData());
  },
});

export {Favorites};

export default connect(mapStateToProps, mapDispatchToProps)(Favorites);
