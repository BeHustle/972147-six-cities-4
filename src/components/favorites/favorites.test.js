import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import {applyMiddleware, createStore} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import {createAPI} from '../../api/api.js';
import {AuthStatus} from '../../constants.js';
import reducer from '../../reducer/reducer.js';
import {setAuthStatus, setUserInfo} from '../../reducer/user/user.reducer.js';
import {cities} from '../../test-mocks/cities.js';
import {offers} from '../../test-mocks/offers.js';
import {userInfo} from '../../test-mocks/user.js';
import {Favorites} from './favorites.jsx';
import {Router} from 'react-router-dom';
import {history} from '../../history.js';

const api = createAPI();

const store = createStore(
    reducer,
    composeWithDevTools(
        applyMiddleware(thunk.withExtraArgument(api))
    )
);

store.dispatch(setUserInfo(userInfo));
store.dispatch(setAuthStatus(AuthStatus.AUTH));

it(`Render favorites`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <Provider store={store}>
            <Favorites
              cities={cities}
              offers={offers}
              onFavoriteClick={() => {}}
              onFavoritesDidMount={() => {}} />
          </Provider>
        </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
