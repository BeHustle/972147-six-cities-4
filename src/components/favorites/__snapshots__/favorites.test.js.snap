// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render favorites 1`] = `
<div
  className="page 4"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  user@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Amsterdam
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Beautiful & luxurious apartment at great locatccion"
                      className="place-card__image"
                      height="110"
                      src="img/apartment-01.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        132
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="#"
                    >
                      Beautiful & luxurious apartment at great locatccion
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    hotel
                  </p>
                </div>
              </article>
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Wood and stone place"
                      className="place-card__image"
                      height="110"
                      src="img/apartment-02.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        254
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button false"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="#"
                    >
                      Wood and stone place
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    apartment
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Paris
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Canal View Prinsengracht"
                      className="place-card__image"
                      height="110"
                      src="img/apartment-03.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        178
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "60%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="#"
                    >
                      Canal View Prinsengracht
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    room
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Cologne
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__card place-card"
              >
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Nice, cozy, warm big bed apartment"
                      className="place-card__image"
                      height="110"
                      src="img/room.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  className="favorites__card-info place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        995
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button false"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        In bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "20%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      href="#"
                    >
                      Nice, cozy, warm big bed apartment
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    house
                  </p>
                </div>
              </article>
            </div>
          </li>
          
          
          
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="/img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
